src/Components/PriceRange/PriceRange.tsx

```diff
@@ -0,0 +1,143 @@
+import { CustomRange, DEFAULT_RANGE } from "Components/PriceRange/constants"
+import { NumericInput } from "Components/PriceRange/NumericInput"
+import { parseRange } from "Components/PriceRange/utils/parseRange"
+import { parseSliderRange } from "Components/PriceRange/utils/parseSliderRange"
+import { convertToFilterFormatRange } from "Components/PriceRange/utils/convertToFilterFormatRange"
+import { getValue } from "Components/PriceRange/utils/getValue"
+
+interface PriceRangeProps {
+  priceRange: string
+  onPriceRangeUpdate: (range: CustomRange) => void
+  bars?: HistogramBarEntity[]
+}
+
+export const PriceRange: FC<PriceRangeProps> = ({
+  priceRange,
+  onPriceRangeUpdate,
+  bars,
+}) => {
+  const [range, setRange] = useState(parseRange(priceRange))
+  const sliderRange = parseSliderRange(range)
+  const [minValue, maxValue] = range
+  const [defaultMinValue, defaultMaxValue] = DEFAULT_RANGE
+
+  useEffect(() => {
+    if (priceRange === "*-*") {
+      setRange(parseRange(priceRange))
+    }
+  }, [priceRange])
+
+  const updateRange = (updatedRange: CustomRange) => {
+    setRange(updatedRange)
+    onPriceRangeUpdate(updatedRange)
+  }
+
+  const handleSliderValueChange = (range: number[]) => {
+    const convertedRange = convertToFilterFormatRange(range)
+
+    updateRange(convertedRange)
+  }
+
+  const handleInputValueChange = (changedIndex: number) => (
+    event: FormEvent<HTMLInputElement>
+  ) => {
+    const { value } = event.currentTarget
+    const updatedRange = range.map((rangeValue, index) => {
+      if (index === changedIndex) {
+        if (value === "" || value === "0") {
+          return "*"
+        }
+
+        return parseInt(value, 10)
+      }
+
+      return rangeValue
+    })
+
+    updateRange(updatedRange)
+  }
+
+  return (
+    <>
+      <Flex>
+        <Box flex={1}>
+          <Text variant="xs" mb={0.5}>
+            Min
+          </Text>
+
+          <NumericInput
+            label="$USD"
+            name="price_min"
+            min="0"
+            step="100"
+            aria-label="Min price"
+            value={getValue(minValue)}
+            onChange={handleInputValueChange(0)}
+          />
+        </Box>
+
+        <Spacer x={2} />
+
+        <Box flex={1}>
+          <Text variant="xs" mb={0.5}>
+            Max
+          </Text>
+
+          <NumericInput
+            label="$USD"
+            name="price_max"
+            min="0"
+            step="100"
+            aria-label="Max price"
+            value={getValue(maxValue)}
+            onChange={handleInputValueChange(1)}
+          />
+        </Box>
+      </Flex>
+
+      <Spacer y={2} />
+
+      {shouldDisplayHistogram(bars) ? (
+        <Histogram
+          bars={bars!}
+          selectedRange={[sliderRange[0], sliderRange[1]]}
+          data-testid="PriceFilterHistogram"
+        />
+      ) : null}
+
+      <Spacer y={shouldDisplayHistogram(bars) ? 2 : 4} />
```

Comment:
**question**: do we really want use different spaces?

Reply:
As for me, it looks way better with bigger Spacer when histogram is absent.

With bigger spacing:
![Screenshot 2023-06-01 at 15 10 48](https://github.com/artsy/force/assets/3934579/3bb850bf-347b-44bf-9392-24beacab33c3)

Without:
![Screenshot 2023-06-01 at 15 10 59](https://github.com/artsy/force/assets/3934579/bf1fd3e8-b2b0-4d7b-87ba-c52a0d8fc2c8)

Reply:
maybe in this case it is better to always use `<Spacer y={4} />`?